### DISCUSSION TITLE
How to add image watermark?

### DISCUSSION THREAD

**Discussion by zn** [SixLabors/ImageSharp#1510] [Q&A] [answered]
Hi! I found [How to add image watermark using SixLabors Imagesharp package in .NET core](https://stackoverflow.com/questions/58988842/how-to-add-image-watermark-using-sixlabors-imagesharp-package-in-net-core) question on stackoverflow, but there are no answers. 
I saw [this sample](https://github.com/SixLabors/Samples/blob/master/ImageSharp/DrawWaterMarkOnImage/Program.cs) of how to add a watermark, but there is a text-based watermark. What I want is to add an image-based watermark. Can you please help me with it?

**P.S.** I've found [similar](https://github.com/SixLabors/ImageSharp/issues/1056) issue, but there are no answers as well

**Comment by JimBobSquarePants** [comment, marked as answer]
Yeah... That's pretty old, We didn't have a dedicated QA then that was persistent.

You can use the [`DrawImage()`](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Processing.DrawImageExtensions.html#SixLabors_ImageSharp_Processing_DrawImageExtensions_DrawImage_SixLabors_ImageSharp_Processing_IImageProcessingContext_SixLabors_ImageSharp_Image_System_Single_) extension for this.

There's a bunch of overloads available to provide different options.

**Reply by zn** [reply]
Thank you!
